editor/editor_properties.cpp

```diff
@@ -913,20 +913,32 @@ void EditorPropertyLayersGrid::_on_hover_exit() {
 		expand_hovered = false;
 		queue_redraw();
 	}
-	if (hovered_index != -1) {
-		hovered_index = -1;
+	if (hovered_index != INT32_MAX) {
+		hovered_index = INT32_MAX;
 		queue_redraw();
 	}
 }
 
-void EditorPropertyLayersGrid::_update_flag() {
-	if (hovered_index >= 0) {
+void EditorPropertyLayersGrid::_update_flag(bool p_replace) {
+	if (hovered_index != INT32_MAX) {
 		// Toggle the flag.
 		// We base our choice on the hovered flag, so that it always matches the hovered flag.
-		if (value & (1 << hovered_index)) {
-			value &= ~(1 << hovered_index);
+		if (p_replace) {
+			// Replace all flags with the hovered flag ("solo mode"),
+			// instead of toggling the hovered flags while preserving other flags's state.
+			if (value == uint32_t(1 << hovered_index)) {
+				// If the flag is already enbled, enable all other items and disable the current flag.
```

Comment:
```suggestion
				// If the flag is already enabled, enable all other items and disable the current flag.
```
If recently added that one to the codespell dict :)